use dequantize per channel group for embedding

manuelcandales opened the pull request:
Summary:
- allow optional zero points on dequantizer_per_channel_group
- add type hints to quantized_ops
- use dequantize per channel group in embedding byte decomposition

Differential Revision: D54813256

jerryzh168 commented on examples/models/llama2/quantize.py:
```diff
@@ -571,10 +571,9 @@ def dequantize_per_channel_group(
-    w_dq = (
-        w_int8_grouped.sub(zero_points).mul(scales).reshape_as(w_int8).to(output_dtype)
-    )
+    zp = zero_points.reshape(-1, 1) if zero_points is not None else 0
```
for `else` does zp have to match the size of scales?

jerryzh168 commented on examples/models/llama2/quantize.py:
```diff
@@ -571,10 +571,9 @@ def dequantize_per_channel_group(
-        w_int8_grouped.sub(zero_points).mul(scales).reshape_as(w_int8).to(output_dtype)
-    )
+    zp = zero_points.reshape(-1, 1) if zero_points is not None else 0
+    w_dq = w_int8_grouped.sub(zp).mul(scales).reshape_as(w_int8).to(output_dtype)
```
`zp` is not used right now btw, can you add a TODO here so we don't forget to add it

jerryzh168 reviewed (approved):
this is great, thanks!

manuelcandales committed:
use dequantize per channel group for embedding (#2374)

Summary:

- add type hints to quantized_ops
- use dequantize per channel group in embedding byte decomposition

bypass-github-export-checks

Reviewed By: mikekgfb

Differential Revision: D54813256
